lib/CType.ml

```diff
@@ -100,37 +103,45 @@ let rec is_pte t = match t with
 
 let rec is_array = function
   | Array _ -> true
-  | Atomic t|Volatile t -> is_array t
-  | _ -> false
+  | Atomic t | Volatile t | Const t -> is_array t
+  | Base _ | Pointer _ -> false
 
 let rec is_atomic = function
-  | Volatile t -> is_atomic t
+  | Volatile t | Const t -> is_atomic t
   | Atomic _ -> true
-  | _ -> false
-
-let strip_atomic0 = function
-  | Atomic t -> t
-  | t -> t
+  | Array _ | Base _ | Pointer _ -> false
 
 let rec strip_atomic = function
   | Volatile t -> Volatile (strip_atomic t)
-  | t ->  strip_atomic0 t
-
-let strip_volatile0 = function
-  | Volatile t -> t
-  | t -> t
+  | Atomic t -> strip_atomic t (* We handle the case where we have malformed types *)
```

Comment:
I am not sure I understand the comment. Also this changes what strip_atomic does, `strip_atomic (Atomic t)` used to return t now it will return strip_atomic t. In most cases that should be fine but I don't see why we want to do it this way.

Reply:
I think my reasoning was if _somehow_ a user provides a nested type like `Atomic (Atomic t)` we would catch it

Reply:
I guess this is what I meant by malformed

Reply:
I think this should never happen. We shouldn't ever create a type Atomic (Atomic t) or Volatile (Volatile t), ideally we should catch these as parser errors. I think this for "compound" types like Const (Array t), Volatile (Pointer t) which make sense in C/C++.